tidy3d/plugins/autograd/functions.py

```diff
@@ -581,3 +582,57 @@ def morphological_gradient_external(
         The external morphological gradient of the input array.
     """
     return grey_dilation(array, size, structure, mode=mode, maxval=maxval) - array
+
+
+def rescale(
+    array: np.ndarray, out_min: float, out_max: float, in_min: float = 0.0, in_max: float = 1.0
+) -> np.ndarray:
+    """
+    Rescale an array from an arbitrary input range to an arbitrary output range.
+
+    Parameters
+    ----------
+    array : np.ndarray
+        The input array to be rescaled.
+    out_min : float
+        The minimum value of the output range.
+    out_max : float
+        The maximum value of the output range.
+    in_min : float, optional
```

Comment:
our convention (which might not be standard) is to use `: type = default value` for optional types, eg `float = 0.0`.

Reply:
yea, we kind of broke the convention a while ago, there's a function that parses these in the base model for our docs API and so it expects the new convention. I'm not totally opposed to just sticking with standard numpydoc  but someone's gotta go through and do the conversions..

Reply:
https://github.com/flexcompute/tidy3d/blob/develop/tidy3d/components/base.py#L875-L939

Reply:
oh wait, ignore that, it's for pydantic models.. I'm actually not sure anymore how these docstrings get rendered for our docs..